src/components/mongo.ts

```diff
@@ -0,0 +1,409 @@
+        cpu: String(cpu),
+        memory: String(memory),
+      });
+      if (typeof size === 'string') {
+        return mapCapabilities(PredefinedSize[size]);
+      }
+      if (typeof size === 'object') {
+        return mapCapabilities(size);
+      }
+      throw Error('Incorrect EcsService size argument');
+    });
+
+    const taskDefinition = new aws.ecs.TaskDefinition(
+      `${this.name}-task-definition`,
+      {
+        family: `${this.name}-task-definition-${stack}`,
+        networkMode: 'awsvpc',
+        executionRoleArn: taskExecutionRole.arn,
+        taskRoleArn: taskRole.arn,
+        cpu: parsedSize.cpu,
+        memory: parsedSize.memory,
+        requiresCompatibilities: ['FARGATE'],
+        containerDefinitions: pulumi
+          .all([
+            this.name,
+            argsWithDefaults.image,
+            argsWithDefaults.port,
+            argsWithDefaults.environment,
+            argsWithDefaults.secrets,
+            this.logGroup.name,
+            awsRegion,
+          ])
+          .apply(
+            ([
+              containerName,
+              image,
+              port,
+              environment,
+              secrets,
+              logGroup,
+              region,
+            ]) => {
+              return JSON.stringify([
+                {
+                  readonlyRootFilesystem: false,
+                  name: containerName,
+                  image,
+                  essential: true,
+                  portMappings: [
+                    {
+                      containerPort: port,
+                      protocol: 'tcp',
+                    },
+                  ],
+                  mountPoints: [
+                    {
+                      containerPath: '/data/db',
+                      sourceVolume: `${this.name}-volume`,
+                    },
+                  ],
+                  logConfiguration: {
+                    logDriver: 'awslogs',
+                    options: {
+                      'awslogs-group': logGroup,
+                      'awslogs-region': region,
+                      'awslogs-stream-prefix': 'ecs',
+                    },
+                  },
+                  environment,
+                  secrets,
+                },
+              ] as ContainerDefinition[]);
+            },
+          ),
+        volumes: [
+          {
+            name: `${this.name}-volume`,
+            efsVolumeConfiguration: {
+              fileSystemId: this.mountTarget.fileSystemId,
+              transitEncryption: 'ENABLED',
+            },
+          },
+        ],
+        tags: { ...commonTags, ...argsWithDefaults.tags },
+      },
+      { parent: this },
+    );
+
+    return taskDefinition;
+  }
+
+  private createServiceDiscovery(args: MongoArgs) {
+    const privateDnsNamespace = new aws.servicediscovery.PrivateDnsNamespace(
+      `${this.name}-private-dns-namespace`,
+      {
+        vpc: args.vpc.vpcId,
```

Comment:
Maybe it is also a good idea to set a friendly namespace name here instead of using a generated one. It can be set to serviceName for example.